### PR TITLE
[refactor embeddings] gligen + ip-adapter

### PR DESCRIPTION
**Pull request by yiyixuxu**
continue refactoring embeddings: GLIGEN + IP-Adapter

### CODE REVIEW

File: src/diffusers/pipelines/deprecated/versatile_diffusion/modeling_text_unet.py [hunk 1]
```diff
@@ -267,7 +267,6 @@ def forward(
         return objs
 
 
-# Copied from diffusers.models.unet_2d_condition.UNet2DConditionModel with UNet2DConditionModel->UNetFlatConditionModel, nn.Conv2d->LinearMultiDim, Block2D->BlockFlat
```
Comment (yiyixuxu) [inline src/diffusers/pipelines/deprecated/versatile_diffusion/modeling_text_unet.py:270]: @DN6 @patrickvonplaten 
do we intend to keep all the `#copied from` statements for deprecated pipelines? that means we still have to maintain them, no?

File: src/diffusers/models/embeddings.py [hunk 2]
```diff
@@ -620,29 +620,34 @@ def shape(x):
...
+    emb = emb.permute(0, 1, 3, 4, 2).reshape(batch_size, num_boxes, embed_dim * 2 * 4)
 
-    def __call__(self, x):
-        x = self.freq_bands * x.unsqueeze(-1)
-        return torch.stack((x.sin(), x.cos()), dim=-1).permute(0, 1, 3, 4, 2).reshape(*x.shape[:2], -1)
+    return emb
 
 
-class PositionNet(nn.Module):
+class GLIGENTextBoundingboxProjection(nn.Module):
```
Comment (yiyixuxu) [inline src/diffusers/models/embeddings.py:644]: @patrickvonplaten
Any rules around when we call it "Projection" vs "Embedding"?

**Comment by yiyixuxu** [comment]
The remote_code tests fail due to the class name change. What should we do about this?

File: src/diffusers/models/embeddings.py (hunk 2 above)
Comment (sayakpaul) [reply src/diffusers/models/embeddings.py:644]: I follow this rule of thumb. 

When you're representing some feature (time, bbox coordinates, tokens, etc.) in the latent space for the first time, it's better to call those representations embeddings. The subsequent operations performed on those representations are projections (more aligned with the literature in linear algebra).

**Comment by sayakpaul** [comment]
> The remote_code tests fail due to the class name change. What should we do about this?

1. We could define a mapping at the import level and manually check for them. 
2. Introduce this as a breaking change.

File: src/diffusers/models/embeddings.py (hunk 2 above)
Comment (patrickvonplaten) [reply src/diffusers/models/embeddings.py:644]: I like `GLIGENTextBoundingboxProjection` - it's nicely specific

**Review by patrickvonplaten — APPROVED** [review]
Very nice refactor - good to have more specific names here

**Comment by DN6** [comment]
@yiyixuxu  I think it's okay to introduce as a breaking change. I don't seem to see it imported directly in a lot of public repos
https://github.com/search?q=%22from+diffusers.models.embeddings+import+PositionNet%22+language:Python&type=code

And importing the GLIGEN pipeline will be unaffected.

**Review by DN6 — APPROVED** [review]
LGTM 👍🏽

**Pull request merged**